url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/map/ArgMappingBuilderUtil.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Scripty Programming Language
+ * Copyright (C) 2010-2012 Bruno Ranschaert, S.D.I.-Consulting BVBA
+ * http://www.sdi-consulting.be
+ * mailto://info@sdi-consulting.be
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.sdicons.scripty.spec.map;
+
+import com.sdicons.scripty.parser.IContext;
+import com.sdicons.scripty.parser.IEval;
+import com.sdicons.scripty.annot.ScriptyBindingParam;
+import com.sdicons.scripty.annot.ScriptyParam;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+public class ArgMappingBuilderUtil
+{
+    public static ArgListMapping buildArgMapping(Method aMethod, Map<String, IArgMapping> aMappings)
+    throws ArgMappingException
+    {
+        Class<?>[] lParamTypes = aMethod.getParameterTypes();
+        Annotation[][] lParamAnnotations = aMethod.getParameterAnnotations();
+        
+        ArgListMapping lArgListMapping = new ArgListMapping();
+        
+        for(int i = 0; i < lParamTypes.length; i++)
+        {
+            IArgMapping lMapping = buildMapping(lParamTypes[i], lParamAnnotations[i], aMappings);
+            lArgListMapping.addArgMapping(lMapping);
+        }
+        return lArgListMapping;
+    }
+    
+    private static IArgMapping buildMapping(Class<?> aParamClass, Annotation[] aParamAnnotations, Map<String, IArgMapping> aMappings)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.